ambry-api/src/test/java/com.github.ambry/account/AccountContainerTest.java

```diff
@@ -176,48 +190,42 @@ public void testMismatchForAccountId() {
-    badMetadata.remove(ACCOUNT_ID_KEY);
-    creatAccountWithBadJsonMetadataAndFail(badMetadata, JSONException.class);
+    JSONObject badMetadata2 = deepCopy(refAccountJson);
+    badMetadata2.remove(ACCOUNT_ID_KEY);
+    TestUtils.assertException(JSONException.class, () -> Account.fromJson(badMetadata2), null);
 
     // unsupported account json version
-    badMetadata = new JSONObject(accountJsonLike, JSONObject.getNames(accountJsonLike));
-    badMetadata.put(JSON_VERSION_KEY, 2);
-    creatAccountWithBadJsonMetadataAndFail(badMetadata, IllegalStateException.class);
+    JSONObject badMetadata3 = deepCopy(refAccountJson).put(Account.JSON_VERSION_KEY, 2);
+    TestUtils.assertException(IllegalStateException.class, () -> Account.fromJson(badMetadata3), null);
 
     // invalid account status
-    badMetadata = new JSONObject(accountJsonLike, JSONObject.getNames(accountJsonLike));
-    badMetadata.put(STATUS_KEY, "invalidAccountStatus");
-    creatAccountWithBadJsonMetadataAndFail(badMetadata, IllegalArgumentException.class);
+    JSONObject badMetadata4 = deepCopy(refAccountJson).put(Account.STATUS_KEY, "invalidAccountStatus");
+    TestUtils.assertException(IllegalArgumentException.class, () -> Account.fromJson(badMetadata4), null);
 
     // null container metadata
-    creatContainerWithBadJsonMetadataAndFail(null, IllegalArgumentException.class);
+    TestUtils.assertException(IllegalArgumentException.class, () -> Container.fromJson(null, refAccountId), null);
 
     // invalid container status
-    badMetadata = new JSONObject(containerJsonLikeList.get(0), JSONObject.getNames(containerJsonLikeList.get(0)));
-    badMetadata.put(Container.STATUS_KEY, "invalidContainerStatus");
-    creatContainerWithBadJsonMetadataAndFail(badMetadata, IllegalArgumentException.class);
+    JSONObject badMetadata5 = deepCopy(v1ContainerJsonList.get(0)).put(Container.STATUS_KEY, "invalidContainerStatus");
+    TestUtils.assertException(IllegalArgumentException.class, () -> Container.fromJson(badMetadata5, refAccountId),
+        null);
 
     // required fields are missing.
-    badMetadata = new JSONObject(containerJsonLikeList.get(0), JSONObject.getNames(containerJsonLikeList.get(0)));
-    badMetadata.remove(CONTAINER_ID_KEY);
-    creatContainerWithBadJsonMetadataAndFail(badMetadata, JSONException.class);
+    JSONObject badMetadata6 = deepCopy(v1ContainerJsonList.get(0));
+    badMetadata6.remove(CONTAINER_ID_KEY);
+    TestUtils.assertException(JSONException.class, () -> Container.fromJson(badMetadata6, refAccountId), null);
 
     // unsupported container json version
-    badMetadata = new JSONObject(containerJsonLikeList.get(0), JSONObject.getNames(containerJsonLikeList.get(0)));
-    badMetadata.put(Container.JSON_VERSION_KEY, 2);
-    creatContainerWithBadJsonMetadataAndFail(badMetadata, IllegalStateException.class);
+    JSONObject badMetadata8 = deepCopy(v1ContainerJsonList.get(0)).put(Container.JSON_VERSION_KEY, 3);
```

Comment:
minor: why there is no badMetadata7 ? ;)

Reply:
fixed, must've removed a test that was no longer applicable.